webapp/src/components/sidebar_right/index.tsx

```diff
@@ -0,0 +1,255 @@
+        />);
+}
+
+function mapGitlabItemListToPrList(gilist: Item[]) {
+    if (!gilist) {
+        return [];
+    }
+
+    return gilist.map((pr: Item) => {
+        return {sha: pr.sha, project_id: pr.project_id, iid: pr.iid};
+    });
+}
+
+function shouldUpdateDetails(prs: Item[], prevPrs: Item[], targetState: string, currentState: string) {
+    if (currentState !== targetState) {
+        return false;
+    }
+
+    if (prs.length !== prevPrs.length) {
+        return true;
+    }
+
+    for (let i = 0; i < prs.length; i++) {
+        if (prs[i].id !== prevPrs[i].id) {
+            return true;
+        }
+    }
+    return false;
+}
+
+function mapPrsToDetails(prs: Item[], details: Item[]) {
+    if (!prs || !prs.length) {
+        return [];
+    }
+
+    return prs.map((pr) => {
+        const foundDetails = details && details.find((prDetails) => pr.project_id === prDetails.project_id && pr.sha === prDetails.sha);
+        if (!foundDetails) {
+            return pr;
+        }
+
+        return {
+            ...pr,
+            status: foundDetails.status,
+            approvers: foundDetails.approvers,
+            total_reviewers: pr.reviewers.length,
+        };
+    });
+}
+
+function SidebarRight({theme}: {theme: Theme}) {
+    const {username, yourAssignments, org, unreads, gitlabURL, rhsState, reviews, yourPrs, reviewDetails, yourPrDetails, getUserLocale} = useSelector((state: GlobalState) => {
+        const store = getPluginState(state);
+        return {
+            username: store.username,
+            reviews: store.reviews,
+            reviewDetails: store.reviewDetails,
+            yourPrs: store.yourPrs,
+            yourPrDetails: store.yourPrDetails,
+            yourAssignments: store.yourAssignments,
+            unreads: store.unreads,
+            org: store.organization,
+            gitlabURL: store.gitlabURL,
+            rhsState: store.rhsState,
+            getUserLocale: getCurrentUserLocale(state),
+        };
+    });
+
+    // States used for storing the PRs/Reviews along with their details which are present separately in redux state.
+    const [prsWithDetails, setPrsWithDetails] = useState<Item[]>(yourPrs);
+    const [reviewsWithDetails, setReviewsWithDetails] = useState<Item[]>(reviews);
+
+    const dispatch = useDispatch();
+
+    useEffect(() => {
+        setReviewsWithDetails(mapPrsToDetails(reviews, reviewDetails));
+    }, [reviews, reviewDetails]);
+
+    useEffect(() => {
+        setPrsWithDetails(mapPrsToDetails(yourPrs, yourPrDetails));
+    }, [yourPrs, yourPrDetails]);
+
+    useEffect(() => {
+        if (yourPrs && rhsState === RHSStates.PRS) {
+            dispatch(getYourPrDetails(
+                mapGitlabItemListToPrList(yourPrs),
+            ));
+            if (reviews && rhsState === RHSStates.REVIEWS) {
+                dispatch(getReviewDetails(
+                    mapGitlabItemListToPrList(reviews),
+                ));
+            }
+        }
```

Comment:
These shouldn't be nested since they are two different cases that will never happen at the same time

```suggestion
        if (yourPrs && rhsState === RHSStates.PRS) {
            dispatch(getYourPrDetails(
                mapGitlabItemListToPrList(yourPrs),
            ));
        }
        
        if (reviews && rhsState === RHSStates.REVIEWS) {
            dispatch(getReviewDetails(
                mapGitlabItemListToPrList(reviews),
            ));
        }
```